Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -1294,43 +1331,44 @@ variable {R M M₂}
 `M₂`, construct the corresponding multilinear map on `n+1` variables obtained by concatenating
 the variables, given by `m ↦ f (init m) (m (last n))`-/
 def MultilinearMap.uncurryRight
-    (f : MultilinearMap R (fun i : Fin n => M i.cast_succ) (M (last n) →ₗ[R] M₂)) :
+    (f : MultilinearMap R (fun i : Fin n => M (castSucc i)) (M (last n) →ₗ[R] M₂)) :
     MultilinearMap R M M₂ where
   toFun m := f (init m) (m (last n))
-  map_add' dec m i x y := by
-    rw [Subsingleton.elim dec (by infer_instance)]
+  map_add' := @fun dec m i x y => by
+    -- porting note: `clear` not necessary in Lean 3 due to not being in the instance cache
+    rw [Subsingleton.elim dec (by clear dec; infer_instance)]; clear dec
     by_cases h : i.val < n
     · have : last n ≠ i := Ne.symm (ne_of_lt h)
-      rw [update_noteq this, update_noteq this, update_noteq this]
+      simp_rw [update_noteq this]
       revert x y
-      rw [(cast_succ_cast_lt i h).symm]
+      rw [(castSucc_cast_lt i h).symm]
```

Comment:
`castSucc_cast_lt` is misnamed, it should be `castSucc_castLt`. I wonder why mathport can't align it.

Reply:
#3320

Reply:
Done. Is this ready to merge?